widgets/src/widget_with_image.rs

```diff
@@ -0,0 +1,74 @@
+use {
+    crate::{
+        makepad_image_formats::*,
+        makepad_draw::*,
+    }
+};
+
+pub trait WidgetWithImage {
+    fn image_filename(&self) -> &LiveDependency;
+    fn texture(&mut self) -> &mut Option<Texture>;
```

Comment:
NIT: I think we could make it a bit clearer with making the name of the trait describe specifically what behaviour it adds to the implementer.
Something like ImageRenderingWidget or ImageLoadingWidget